lib/vdsm/tool/configurators/multipath.py

```diff
@@ -109,3 +106,27 @@ def isconfigured():
 
     sys.stdout.write("multipath requires configuration\n")
     return NO
+
+
+def isconfigured():
+    """
+    Check the multipath daemon configuration.
+    """
+    conf_metadata_ok = _check_mpath_metadata()
+    try:
+        mpathconf.check_local_config()
+    except exception.InvalidConfiguration as e:
+        ufn_enabled_section = e.context["section"]
+        sys.stdout.write(
+            f"Invalid configuration detected: {e.context['reason']}:\n")
+        sys.stdout.write(f"  {ufn_enabled_section.name} {{\n")
+        for attr in ufn_enabled_section.attrs:
+            line_indent = "    "
+            if attr.key == "user_friendly_names":
+                line_indent = "->" + line_indent
+            sys.stdout.write(f"{line_indent}{attr.key} {attr.value}\n")
+        sys.stdout.write("  }\n")
+        sys.stdout.write("This configuration is not supported "
+                         "and may lead storage domain corruption.\n")
+        return NO
```

Comment:
Returning NO means that vdsm will not start on a system with unsupported configuration. This may break existing setups so we may want to just warn about the issue without failing vdsm-tool.

We can start by warning, and later convert the warning to hard error if needed.

Reply:
The whole point of the bug fix is making it fail to avoid wrong configurations, no? The issue may get unnoticed if the upgrade tool logs are not checked.

Reply:
Ok. Let's go for it then, and start with a warning. My main concern is in that the warning can be easily missed, but I understand that breaking after an upgrade is not something desirable for users either.